forwards_paper.tex

```diff
@@ -658,43 +624,51 @@ \subsection*{Simulation benchmarks}
+for use in other code bases.
+
+The tree sequences produced by default by this method
+are very compact, storing genotype \emph{and} genealogical information
+in a small fraction of the space taken by a compressed VCF file.
+The format is also highly efficient to process,
+leading to advantages for downstream analysis as well.
+This is because many algorithms to compute statistics of interest for population genetics
+are naturally expressed in terms of tree topologies,
+and so can be quickly computed from the trees underlying the tree sequence format. 
+For example, pairwise nucleotide diversity $\pi$, is the average density of
+differences between sequences in the sample. 
+To compute this directly from sequence data at $m$ sites in $n$ samples
+requires computing allele frequencies, taking $O(nm)$ operations.
+By using the locations of the mutations on the marginal trees,
+and the fact that these are correlated, 
+dynamic programming algorithms similar to those in~\citep{kelleher2016efficient}
+can do this in roughly $O(n + m \log n)$ operations.
 The \msprime\ API provides a method to compute $\pi$ among arbitrary subsets of the
-samples in a tree sequence, and calculating $\pi$ over all 500K samples
-in the example above required about 1.2 seconds.
-\jk{I haven't checked the analysis of computing $\pi$ here and not thought about
-it too deeply. I guess we should also compare this with something else in order to
-show that this time is pretty good. We could get a subset of the sites and push
-the data into pylibseq maybe, or scikit-allel??}
-
-This is a general-purpose strategy that can be applied to other methods.
-
-A hybrid approach, in which the recent history of a large sample of
-individuals is simulated under a detailed forwards-in-time model and the deep history
-under the coalescent, is an attractive prospect.
-\jk{Not sure where this should go; perhaps in the discussion? We can make the point that
-by keeping track of the genealogies forwards in time we can easily combine these with the
-deep history simulated by msprime}
+samples in a tree sequence, which took about 1.2 seconds
+applied to the example tree sequence above with 1.1 million mutations 
+in 200 megabases for 500,000 samples.
+
+Another attractive feature of this set of tools
+is that it makes it easy to incorporate \emph{prior history},
+simply by seeding the simulation with a (relatively inexpensive) coalescent simulation.
+This allows for incorporation of deep-time history beyond the reach of individual-based simulations. 
+Since geographic structure from times longer ago than the mixing
+time of migration across the range has limited effect on modern genealogies
+\citep{wilkins2004separation} (other than possibly changing effective population
+size \citet{barton2002neutral,cox2002stepping}), this may not negatively affect realism.
 
-Parallelization will be important;
-this requires only one-way communication.
```

Comment:
meh

Reply:
I think this will be critical moving forwards! Ftprime takes so much ram that CPU gets wasted on multicore systems.  And the profiling of my code suggests big gains from asynchronous execution of the msprime steps.

Reply:
oh, i think so too!  I just didn't feel like we had to make that point here.